### ISSUE TITLE
Which weight file to use for transfer learning?

### ISSUE THREAD

**Issue by GaiaGrower** [matterport/Mask_RCNN#1284]
I trained the nuclei images from the Segmenting Nuclei in Microscopy Images project, both network heads and all layers, with 10 epochs for each (20 total), using the mask_rcnn_coco.h5 weight file as the
starting place, and a modified version of nucleus.py.

Now I have 20 weight files, created from each of those epochs.
I want to train Mask R-CNN on a different dataset, one of images of soil bacteria.

Which weight file, of the 20, should I use instead of the mask_rcnn_coco.h5 file to start this training session?

Each weight file's epoch has a different set of "losses".  Should I use the very last epoch's weight 
file to start training on my own images?  Or should I choose the weight file from my 20, that has the best set of "losses".

Should I use one from the "heads only" run, or from the "all layers" run?

Thank you for your help.

**Comment by hackerghost93** [comment]
How big is your dataset? I would use coco weights for transfer learning as well and start with the heads only then maybe training all the network.

**Comment by GaiaGrower** [comment]
Hello, and thank you for replying!  I apologize for my ignorance; just starting out.  

I'm using the dataset that came with the nucleus sample, about 30k images all told.  

I'm trying to get a model together that will recognize soil microorganisms: bacteria, fungi, nematodes, and protozoa.  I thought that finding, thank goodness, a dataset of microscope slide images would be a better place to start with, than sets of above ground images.

I thought that the point of training the nucleus model on nuclei images, starting with the coco weights, was to get a new weight file that was the result of the model learning to identify nuclei.  From there, using my nuclei-trained weight file, I could train the mask model to identify, to start, soil bacteria.

Am I totally off the mark?  Why would I use the coco weights again, if I've got a nuclei-trained weight file to train the model on soil bacteria images?

Which of the 20 nuclei-trained weight files should I use?  Best set of losses, the very last weight file, after training all layers?

Thank you in advance for your help.

**Comment by hackerghost93** [comment]
The best set of losses you can find is the last one of the training ( look at tensorboard for val loss to be sure ). You need to use the same configurations. And I would recommend trying both using the last weights files and start at the beginning from coco. 
Training a model is an iterative process that takes many tries to find the best accuracy.

**Comment by GaiaGrower** [comment]
Thank you for responding Ahmed.  Guess there's plenty left for me to learn
about this.
Learning via email doesn't really work.  Any chance of talking with you for
a few minutes?
I can do phone, Skype, or WhatsApp.
Baruch

**Comment by hackerghost93** [comment]
There is a course which helped me a lot Deep learning specialization on Coursera by Andrew Ng. you should really look into it.

**Comment by ectg** [comment]
> The best set of losses you can find is the last one of the training ( look at tensorboard for val loss to be sure ).

Hi @hackerghost93, I am a little confused on how to use tensorboard to visualize the losses. Is sample code available in this download?

**Comment by hackerghost93** [comment]
``tensorboard --logdir path_to_current_dir ``
After you run, by default this repository attaches tensorboard callback to it. The only thing to do is to run this command after you are done with training. ** change current directory to where the model checkout. **
